app/src/components/image/image.jsx

```diff
@@ -0,0 +1,194 @@
+import { useEffect, useState } from "react"
+import styles from "./image.module.css"
+
+import Cookies from "universal-cookie"
+
+async function getImageUrl (id) {
+    const url = `${import.meta.env.VITE_BACKEND_URL}/generate/${id}`
+
+    let options = {
+        method: "GET",
+        headers: {
+            'accept': 'application/json',
+            "Content-Type": "application/json",
+        }
+    }
+
+    const response = await fetch(url, options)
+    const resJson = await response.json()
+
+    return resJson
+}
+
+/*
+    Function to manually add Image to Firebase storage
+*/
+async function addImageManually (id, image, user, prompt ) {
+    const url = `${import.meta.env.VITE_BACKEND_URL}/generate/upload-firebase`
+
+    let options = {
+        method: "POST",
+        headers: {
+            'accept': 'application/json',
+            "Content-Type": "application/json",
+        },
+        body : JSON.stringify({
+            'imageUrl' : image,
+            'imageId' : id,
+            'imagePrompt' : prompt,
+            'userId' : user
+        })
+    }
+
+    const response = await fetch(url, options)
+    const resJson = await response.json()
+
+    return resJson
+}
+
+
+
+
+export default function  ArtImage(props) {
+    const [imageUrl, setImageUrl] = useState("")
+    const [allData, setAllData] = useState({})
+    const [generatedData, setGenData] = useState({})
+    const [selected, setSelected] = useState(false)
+
+    const imageData = props.imageData;
+
+    let waitTimeOut;
+    let countingTimeout;
+
+
+    // might keep this state to show remaining time
+    // TODO: replace with loading later
+    const [waitTime, setWaitTime] = useState(Infinity)
+
+    const cookies = new Cookies(null, { path : "/"})
+
+    /*
+        Recursive function to get firebase image url
+        Only called when image is still processing
+    */
+    async function setTimeoutFunction (wait) {
+        const data = await getImageUrl(imageData.id)
+        if (data.error) {return}
+
+        // TODO: if we get generated image (response from ai horde), insert into database manually
+        if (data.imgUrl) {
+            setAllData(data)
+            setImageUrl(data.imgUrl)
+
+            clearTimeout(waitTimeOut)
+            clearInterval(countingTimeout)
+        }else if (data.generations.length > 0) {
+            clearTimeout(waitTimeOut)
+            clearInterval(countingTimeout)
+            setImageUrl(data.generations[0].img)
+            setWaitTime(0)
+
+            // set All data after adding to prisma and firebase
+            const tempData = {
+                'id' : data.generations[0].id,
+                'imgUrl' : data.generations[0].img,
+                'userId' : imageData.userId,
+                'prompt' : imageData.imagePrompt
+            }
+
+            setGenData({...data.generations[0], ...tempData})
+
+        } else{
+            waitTimeOut = setTimeout(async () => {
```

Comment:
@shaynak This routine waits until the image is done 'cooking' before fetching the url. It only displays the wait-time to the user, I would've liked a timer but the AI I'm using is pretty inconsistent, it can go from 110 seconds to 120 seconds, and then back to 80 seconds. Any ideas?